BitmovinAnalyticsCollector/Classes/AVPlayer/Classes/AVPlayerAdapter.swift

```diff
@@ -81,20 +86,38 @@ class AVPlayerAdapter: NSObject, PlayerAdapter {
 
             break
         case .failed:
-            stateMachine.transitionState(destinationState: .error, time: player?.currentTime())
+            let error = self.player?.currentItem?.error as NSError?
+            let errorCode = error?.code ?? 1
+            let errorMessage = error?.localizedDescription ?? "Unkown"
+            
+            stateMachine.transitionState(destinationState: .error, time: player?.currentTime(), data: [BitmovinAnalyticsInternal.ErrorCodeKey: errorCode, BitmovinAnalyticsInternal.ErrorMessageKey: errorMessage])
             break
         default:
             break
         }
     }
 
+    @objc private func failedToPlayToEndTime(notification: Notification) {
+        let error = notification.userInfo?[AVPlayerItemFailedToPlayToEndTimeErrorKey] as? NSError
+        
+        let errorCode = error?.code ?? 1
```

Comment:
Is it the error code that we are using in case there in no error object whatsoever? @Tigraine

Reply:
Yeah we have to set something, so filtering for errors on the API side works. And as some old code uses ERROR_CODE > 0, I decided 1 is a good choice :D